### ISSUE TITLE
Support for v3 scheduler

### ISSUE THREAD

**Issue by user1823** [ankidroid/Anki-Android#10411]
The Anki 2021 scheduler (v3) is supported on both Anki Desktop (2.1.45+) and AnkiMobile (2.0.75+). So, kindly add the support to AnkiDroid for compatibility.

**Comment by mikehardy** [comment]
Hi there! I believe this will require newer database schemas to be supported first, which will mean completion of this issue at least: https://github.com/ankidroid/Anki-Android-Backend/issues/1

Note that all available developers that could do this at the moment are basically working with all available time to either finish the scoped storage migration so we may release 2.16 - it could be quite some time (months, many months, a year even) before this happens

**Comment by mikehardy** [comment]
Hey everyone that sees this! V3 Scheduler is ready for testing, and we have made it available for regular users.

Please make sure you have a backup first just in case! 

Here's what to do:

1. Backup: copy `/sdcard/AnkiDroid/collection.anki2` somewhere safe. I use [the file manager X-Plore](https://play.google.com/store/apps/details?id=com.lonelycatgames.Xplore) for this personally, but any file manager will let you do this. Now you have a backup. Just in case.
2. Install our alpha release (must be newer than 2.16alpha78 if you already have the alpha update it): https://github.com/ankidroid/Anki-Android/releases
3. Go to "Settings" --> "Advanced" and enable "Use the new backend" in "Experimental Settings"

At this point, if your collection was set to v3 *on Anki Desktop*, AnkiDroid should *also* use the same v3 scheduler code for scheduling.

We do not currently have a way within AnkiDroid to turn on the v3 scheduler so you have to enable that in Anki Desktop then sync

Please open new issues here if you see anything wrong, and we will examine. Thanks!

[this comment is subject to edit as I get confirmation all the information here is technically correct, thanks for your patience, and thanks for testing!]

**Comment by mikehardy** [comment]
Updated the instructions above, the advanced settings pane now lets you enable v3 scheduler directly, and no app restart is required. Everything *seems* to be working...

**Comment by SnipeAT** [comment]
I'm testing the crap outta it!

**Comment by santiyounger** [comment]
can't wait to update ankidroid to V3, thanks a lot! What are potential risks of using this alpha version with v3 scheduler? has anyone found any limitations? is it safe to use it, or is it just for testing?

Thanks for making Anki in android a great experience

**Comment by mikehardy** [comment]
We have had a few crashes for people using the new-backend / v3 scheduler combo, but when I say "few" I think the count is...3? or 4? And typically we are able to fix them within a couple days + they only occur with specific actions (which you may avoid in the meantime). That is actually a huge service to us as well, as it allows us to hear about + locate + fix the problems early while just a few people are testing, so from our perspective it's a nice way for non-developers to help us, if that's your thing :-)

I do not want to be too bold and say there are no other risks, but as far as we can see, it is all basically working, so should not be *that* risky. We warn people to take a backup just to be 100% sure, just in case. We know some people have huge collections with 10 years worth of data so...why not take a backup.

In general, it should be working well right now then and we would love more testers and reports of anything strange while v3 is in use.

**Comment by shmuelsash** [comment]
Is there a timeline for when this will be out of testing?

**Comment by mikehardy** [comment]
With sincere apologies, I do not answer timelines in open source projects as a general rule. There can be no reliable deadlines in volunteer projects so committing to one just leads to disappointment for users and guilt for volunteers.

It is one of our highest priorities though, I can assure you of that. It's blocked pending this issue, you may follow it if you want updates - when it closes, we'll enter the beta cycle and release stable as soon as possible after #5304

**Comment by santiyounger** [comment]
> In general, it should be working well right now then and we would love more testers and reports of anything strange while v3 is in use.

Thanks @mikehardy  you are the man, will definitely go for it! thank you for your incredible work, if I find any issues is this the best place post them?

**Comment by mikehardy** [comment]
I do work hard on it, but this is a really big team now and lots are working hard to make the new scheduler (and everything else) work, that's for sure :-)

Report them either here or a completely new issue, with a lean towards completely issue, with debug information requested (especially if it's a crash, the "ACRA UUID" - your anonymous crash tag - lets us find the trace so we can fix it

**Comment by santiyounger** [comment]
thanks @mikehardy appreciate it, so far so good. I upgraded to v3 on ankidroid (and on windows) and from what I can tell no issues so far. Also, the new interface for ankidroid is amazing, really enjoying it! Keep it up, amazing work!
